src/recordlinker/linkage/matchers.py

```diff
@@ -0,0 +1,427 @@
+        strings, defaults to "JaroWinkler".
+     - JaroWinkler: a ratio of matching characters and transpositions needed to
+        transform string1 into string2.
+     - Levenshtein: the number of edits (excluding transpositions) needed to transform
+        string1 into string2.
+     - DamerauLevenshtein: the number of edits (including transpositions) needed to
+        transform string1 into string2.
+    :return: The normalized similarity between string1 and string2, with 0 representing
+        no similarity between string1 and string2, and 1 meaning string1 and string2 are
+        dentical words.
+    """
+    if similarity_measure == "JaroWinkler":
+        return rapidfuzz.distance.JaroWinkler.normalized_similarity(string1, string2)
+    elif similarity_measure == "Levenshtein":
+        return rapidfuzz.distance.Levenshtein.normalized_similarity(string1, string2)
+    elif similarity_measure == "DamerauLevenshtein":
+        return rapidfuzz.distance.DamerauLevenshtein.normalized_similarity(
+            string1, string2
+        )
+
+
+def eval_perfect_match(feature_comparisons: list, **kwargs) -> bool:
+    """
+    Determines whether a given set of feature comparisons represent a
+    'perfect' match (i.e. whether all features that were compared match
+    in whatever criteria was specified for them).
+
+    :param feature_comparisons: A list of 1s and 0s, one for each feature
+      that was compared during the match algorithm.
+    :return: The evaluation of whether the given features all match.
+    """
+    return sum(feature_comparisons) == len(feature_comparisons)
+
+
+def eval_log_odds_cutoff(feature_comparisons: list, **kwargs) -> bool:
+    """
+    Determines whether a given set of feature comparisons matches enough
+    to be the result of a true patient link instead of just random chance.
+    This is represented using previously computed log-odds ratios.
+
+    :param feature_comparisons: A list of floats representing the log-odds
+      score of each field computed on.
+    :return: Whether the feature comparisons score well enough to be
+      considered a match.
+    """
+    if "true_match_threshold" not in kwargs:
+        raise KeyError("Cutoff threshold for true matches must be passed.")
+    return sum(feature_comparisons) >= kwargs["true_match_threshold"]
+
+
+def feature_match_exact(
+    record_i: list,
+    record_j: list,
+    feature_col: str,
+    col_to_idx: dict[str, int],
+    **kwargs: dict,
+) -> bool:
+    """
+    Determines whether a single feature in a given pair of records
+    constitutes an exact match (perfect equality).
+
+    :param record_i: One of the records in the candidate pair to evaluate.
+    :param record_j: The second record in the candidate pair.
+    :param feature_col: The name of the column being evaluated (e.g. "city").
+    :param col_to_idx: A dictionary mapping column names to the numeric index
```

Comment:
Out of scope for this PR, but agree w/ past discussions that we can refactor this code to not rely on passing around a `col_to_idx` dict mapping column names to order index. The current implementation is less DRY and readable (which columns are being evaluated?).